src/lib/parquet/src/Flow/Parquet/BinaryReader/BinaryBufferReader.php

```diff
@@ -158,7 +158,7 @@ public function readFloats(int $total) : array
         $floats = [];
 
         foreach ($floatBytes as $bytes) {
-            $floats[] = \unpack($this->byteOrder === ByteOrder::LITTLE_ENDIAN ? 'g' : 'G', \pack('C*', ...$bytes))[1];
+            $floats[] = \round(\unpack($this->byteOrder === ByteOrder::LITTLE_ENDIAN ? 'g' : 'G', \pack('C*', ...$bytes))[1], 7);
```

Comment:
yeah, php is a bit retarded with floats, I can't find a way to write/read floats without losing a precision